feature/security/gnsi/certz/tests/internal/setup_service/setup_service.go

```diff
@@ -0,0 +1,586 @@
+			CreatedOn: varClock,
+			Entity:    &certzpb.Entity_AuthenticationPolicy{AuthenticationPolicy: entityContent.(*certzpb.AuthenticationPolicy)}}
+
+	default:
+		t.Fatalf("Invalid entity type")
+	}
+	return certzpb.Entity{}
+}
+
+// CreateCertzChain function to get the certificate chain of type certificate chain/trust bundle.
+func CreateCertzChain(t *testing.T, certData CertificateChainRequest) certzpb.CertificateChain {
+
+	switch certData.RequestType {
+	case EntityTypeCertificateChain:
+		if len(certData.ServerCertFile) == 0 {
+			t.Fatalf("Missing server certificate file for creating certificate chain object.")
+		}
+		serverCertContent, err := os.ReadFile(certData.ServerCertFile)
+		if err != nil {
+			t.Fatalf("Error reading Server Certificate file at: %v with error: %v", certData.ServerCertFile, err)
+
+		}
+		if len(certData.ServerKeyFile) != 0 {
+			serverKeyContent, err := os.ReadFile(certData.ServerKeyFile)
+			if err != nil {
+				t.Fatalf("Error reading Server Key file at: %v with error: %v", certData.ServerKeyFile, err)
+			}
+			return certzpb.CertificateChain{Certificate: &certzpb.Certificate{
+				Type:        certzpb.CertificateType_CERTIFICATE_TYPE_X509,
+				Encoding:    certzpb.CertificateEncoding_CERTIFICATE_ENCODING_PEM,
+				Certificate: serverCertContent,
+				PrivateKey:  serverKeyContent}, Parent: nil}
+		}
+		return certzpb.CertificateChain{Certificate: &certzpb.Certificate{
+			Type:        certzpb.CertificateType_CERTIFICATE_TYPE_X509,
+			Encoding:    certzpb.CertificateEncoding_CERTIFICATE_ENCODING_PEM,
+			Certificate: serverCertContent,
+			PrivateKey:  nil}, Parent: nil}
+
+	case EntityTypeTrustBundle:
+		if len(certData.TrustBundleFile) == 0 {
+			t.Fatalf("Missing trust bundle file for creating certificate chain object.")
+		}
+		trustBundleContent, err := os.ReadFile(certData.TrustBundleFile)
+		if err != nil {
+			t.Fatalf("Error reading trust bundle file at: %v with error: %v", certData.TrustBundleFile, err)
+		}
+		return certzpb.CertificateChain{Certificate: &certzpb.Certificate{
+			Type:        certzpb.CertificateType_CERTIFICATE_TYPE_X509,
+			Encoding:    certzpb.CertificateEncoding_CERTIFICATE_ENCODING_PEM,
+			Certificate: trustBundleContent,
+		}, Parent: nil}
+
+	default:
+		t.Fatalf("Invalid request type received.")
+	}
+	return certzpb.CertificateChain{}
+}
+
+// CreateCertChainFromTrustBundle function to create the certificate chain from trust bundle.
+func CreateCertChainFromTrustBundle(fileName string) *certzpb.CertificateChain {
+	pemData, err := os.ReadFile(fileName)
+	if err != nil {
+		return &certzpb.CertificateChain{}
+	}
+	var trust [][]byte
+	for {
+		var block *pem.Block
+		block, pemData = pem.Decode(pemData)
+		if block == nil {
+			break
+		}
+		if block.Type != "CERTIFICATE" {
+			continue
+		}
+		p := pem.EncodeToMemory(block)
+		if p == nil {
+			return &certzpb.CertificateChain{}
+		}
+		trust = append(trust, p)
+	}
+	//a valid check for trust not empty
```

Comment:
// a valid c...

Reply:
almost certainly ALSO make this a proper / full sentence, not a sentence fragment.